### DISCUSSION TITLE
extending prism.js

### DISCUSSION THREAD

**Discussion by basaran** [pngwn/MDsveX#367] [General]
Hello, is the internal `Prism` available to be extended with additional language configuration?

**Comment by pngwn** [comment]
Prism is a global, so the prism instance should be available when you run mdsvex, if you were to write a plugin you should be able to access prism on `globalThis.Prism` or similar. What are you looking to configure?

**Comment by basaran** [comment]
Thank you. I was going to try adding this:

https://github.com/Golmote/prism-treeview

I will try later today, and update here.

**Reply by basaran** [reply]
at a glance, I didn't see the `Prism` inside globalThis, or the `window` of the rendered sveltekit (no SSR).

**Reply by pngwn** [reply]
It will only be present at build time. Prism is not available at runtime on either the client or server.

**Reply by basaran** [reply]
Do you have any ideas how I can insert the Prismjs plugin ?

It's basically calling two methods:

```
Prism.languages.treeview....
```

and 

```
globalThis.Prism.hooks.add('wrap', function(env) {
```

**Reply by basaran** [reply]
I will try to add a prism section to the mdsvex config, and send a PR if I am successful in doing so. I found the `Prism` inside transformers.

**Reply by pngwn** [reply]
All you should really need to do is `import Prism from 'prismjs'` then `import MyLanguage from 'my-prism-language` before running the bundle, so in your svelte config file or vite config file should work. Prism languages attach themselves tot he global Prism instance. There may be some issues with vite dev mode.

Prism won't be used in a future version, so if this doesn't work easily right now, I probably won't be fixing it as  shiki will probably be used in a future version making things simpler.

**Reply by pngwn** [reply]
And example can be found here: https://github.com/pngwn/MDsveX/blob/master/packages/site/rollup.config.js and here https://github.com/pngwn/MDsveX/blob/master/packages/site/prism/prism.js

**Reply by basaran** [reply]
thank you, you're right. This didn't require anything special at all. Simply importing the `Prism` and adding those plugin methods worked out just fine. The magical world of javascript modules :)

![image](https://user-images.githubusercontent.com/30809170/139614291-739cbeea-96d8-4484-8b7f-1214c8798339.png)

So, the reason this did work is because there is only one instance of the module, and by importing it first, I'm intercepting it before you call it. Is that right?

**Reply by trasherdk** [reply]
I would love to see something like this, from ExtJS 3.x :smile: 

![image](https://user-images.githubusercontent.com/5003891/142858031-abd8ae16-2e77-4087-9df3-8e37f5aa022c.png)
